packages/permission-controller/src/PermissionController.ts

```diff
@@ -1181,6 +1191,67 @@ export class PermissionController<
     this.setCaveat(origin, target, caveatType, caveatValue);
   }
 
+  /**
+   * Updates caveat value across all subjects' permissions that have the specified
+   * target and caveat type.
+   *
+   * @template TargetName - The permission target name. Should be inferred.
+   * @template CaveatType - The valid caveat type for the permission. Should
+   * be inferred.
+   * @param target - The target name of the permission.
+   * @param caveatType - The type of the caveat to update.
+   * @param caveatValue - The new value of the caveat.
+   */
+  updateCaveatForAllSubjects<
+    TargetName extends ExtractPermission<
+      ControllerPermissionSpecification,
+      ControllerCaveatSpecification
+    >['parentCapability'],
+    CaveatType extends ExtractAllowedCaveatTypes<ControllerPermissionSpecification>,
+    CaveatValue extends ExtractCaveatValue<
+      ControllerCaveatSpecification,
+      CaveatType
+    >,
+  >(
+    target: TargetName,
+    caveatType: CaveatType,
+    caveatValue: CaveatValue,
+  ): void {
+    const specification = this.getCaveatSpecification(caveatType);
+    const caveat = { type: caveatType, value: caveatValue };
+
+    this.update((draftState) => {
+      Object.entries(draftState.subjects).forEach(([origin, subject]) => {
+        const { permissions } = subject;
+
+        if (hasProperty(permissions, target)) {
+          const caveatIndex = (
+            permissions[target] as PermissionConstraint
+          ).caveats?.findIndex(
+            (existingCaveat) => existingCaveat.type === caveat.type,
+          );
+          if (caveatIndex !== undefined && caveatIndex !== -1) {
+            if (caveatValue === undefined) {
+              throw new CaveatMissingValueError(caveat, origin, target);
+            }
+
+            if (!isValidJson(caveatValue)) {
+              throw new CaveatInvalidJsonError(caveat, origin, target);
+            }
+
+            // Typecast: TypeScript still believes that the caveat is a PlainObject.
+            specification.validator?.(
```

Comment:
Is this meant to throw as well?

Reply:
Yes 😞, also moved down below the two preceding `if` statements because I wanted the validation to be in that order